components/src/core/components/TableFilter/TableFilter.vue

```diff
@@ -44,24 +46,70 @@ export default defineComponent({
       type: String,
       required: true,
     },
+    itemsSelected: {
+      type: Boolean,
+      default: false,
+    },
   },
 
-  setup() {
+  setup(props) {
     const responsiveState = useResponsive();
     const isActive = ref(true);
+    const isSpilled = ref(true);
 
     const toggleFilters = () => {
       isActive.value = !isActive.value;
     };
 
+    const updateIsSpilled = () => {
+      nextTick(function() {
+        const titleWidth = document.getElementsByClassName(
+          'oxd-table-filter-header-title',
+        )[0]
+          ? document.getElementsByClassName('oxd-table-filter-header-title')[0]
+              .clientWidth
+          : 0;
+        const actionButtonWidth = document.getElementsByClassName(
+          'oxd-table-filter-header-bulk-actions',
+        )[0]
+          ? document.getElementsByClassName(
+              'oxd-table-filter-header-bulk-actions',
+            )[0].clientWidth
+          : 0;
+        const fullWidth = document.getElementsByClassName(
+          'oxd-table-filter-header',
+        )[0]
+          ? document.getElementsByClassName('oxd-table-filter-header')[0]
+              .clientWidth
+          : 0;
+        const filterWidth = document.getElementsByClassName(
+          'oxd-table-filter-header-options',
+        )[0]
+          ? document.getElementsByClassName(
+              'oxd-table-filter-header-options',
+            )[0].clientWidth
+          : 0;
+        isSpilled.value =
+          props.itemsSelected &&
+          titleWidth + actionButtonWidth + filterWidth > fullWidth;
+      });
+    };
+
     watchEffect(() => {
       isActive.value =
         responsiveState.screenType === DEVICE_LG ||
         responsiveState.screenType === DEVICE_XL;
     });
+    watchEffect(() => {
+      props.itemsSelected;
+      responsiveState.windowWidth;
+      updateIsSpilled();
+    });
 
+    document.addEventListener('collapsibleViewToggled', updateIsSpilled);
```

Comment:
Please make sure to remove event listener on unmount. can lead to memory leaks

Reply:
@Super-Chama Thanks. I will do that.